creator-node/src/services/ContentNodeInfoManager.ts

```diff
@@ -88,13 +105,259 @@ async function getSpIdFromEndpoint(
   return cNodeEndpointToSpIdMap.get(endpoint)
 }
 
-async function getContentNodeInfoFromSpId(
+export async function getContentNodeInfoFromSpId(
   spId: number,
   logger: Logger,
   redisClient = defaultRedisClient
 ): Promise<ContentNodeFromChain | undefined> {
-  const map = await getMapOfSpIdToChainInfo(logger, redisClient)
-  return map.get(spId)
+  const spIdToChainInfoMap = await getMapOfSpIdToChainInfo(logger, redisClient)
+  return spIdToChainInfoMap.get(spId)
+}
+
+export async function getContentNodeInfoFromEndpoint(
+  endpoint: string,
+  logger: Logger,
+  redisClient = defaultRedisClient
+): Promise<ContentNodeFromChain | undefined> {
+  const endpointToSpIdMap = await getMapOfCNodeEndpointToSpId(
+    logger,
+    redisClient
+  )
+  const spId = endpointToSpIdMap.get(endpoint)
+  if (spId === undefined) return undefined
+  const cNodeInfo = await getContentNodeInfoFromSpId(spId, logger, redisClient)
+  return cNodeInfo
+}
+
+export async function getReplicaSetSpIdsByUserId({
```

Comment:
this is a typed, ever-so-slightly cleaner version of `getReplicaSetSpIDs` from middlewares.js. eventually we should find a way to not have it be such a long, monstrous function. maybe try to separate out the ensurePrimary part

Reply:
yes lol, these functions are so gross